src/Statistics/HistogramTest.cpp

```diff
@@ -166,4 +166,23 @@ TEST(Histogram, BuildHistogramCorrectlyChoosesTheBinWidth) {
   EXPECT_EQ(hist->counts.size(), 128);
 }
 
+static void TestGetBinWidths(int number_of_bins, int histogram_width) {
+  const std::vector<int> widths = GetBinWidth(number_of_bins, histogram_width);
+  ASSERT_EQ(widths.size(), number_of_bins);
+  int sum = std::reduce(std::begin(widths), std::end(widths), 0);
+  ASSERT_EQ(sum, histogram_width);
+  int max = *std::max_element(std::begin(widths), std::end(widths));
+  int min = *std::min_element(std::begin(widths), std::end(widths));
+  ASSERT_TRUE(max - min <= 1);
+  ASSERT_TRUE(std::all_of(std::begin(widths), std::end(widths),
+                          [](const int width) { return width >= 0; }));
```

Comment:
I feel like there is a way to write this with `ASSERT_THAT`. Probably not necessary though.

Reply:
Yes. `ASSERT_THAT(testing::Each(testing::Ge(0)))`

Reply:
It has the advantage that you get better error message in case that check fails - telling which entry failed to match and what has been its value.

Reply:
Wow. Thanks, that's cool.